spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ReactorSleuth.java

```diff
@@ -67,43 +66,37 @@ private ReactorSleuth() {
 			log.trace("Scope passing operator [" + beanFactory + "]");
 		}
 
-		// Adapt if lazy bean factory
-		BooleanSupplier isActive = beanFactory instanceof ConfigurableApplicationContext
-				? ((ConfigurableApplicationContext) beanFactory)::isActive : () -> true;
-
 		return Operators.liftPublisher((p, sub) -> {
-			// if Flux/Mono #just, #empty, #error
+			// While supply of scalar types may be deferred, we don't currently scope
+			// production of values in a trace context. This prevents excessive overhead
+			// when using constant results such as Flux/Mono #just, #empty, #error
 			if (p instanceof Fuseable.ScalarCallable) {
 				return sub;
 			}
-			Scannable scannable = Scannable.from(p);
-			// rest of the logic unchanged...
-			if (isActive.getAsBoolean()) {
+
+			if (beanFactory instanceof ConfigurableApplicationContext
```

Comment:
@adriancole   
before change `instanceof` was calculated only once  
now it is calculated on each `Publisher.subscribe()` and it could be a lot of times  
As far as I understand reactor team trying to avoid "expensive instanceof check" see reactor/reactor-core#1882

Reply:
how is coersing a lambda beanFactory::isActive to boolean supplier memoizing?

Reply:
oh I see it is specifically the instanceof you are trying to cache.. anyway the code is a bit messy and this moved to a place where it is cached anyway. I'll ping you on the next PR

Reply:
actually the logic is a bit strange to me, I guess I messed up because when not an instanceof ConfigurableApplicationContext it should return true (not sure when spring boot will be used without a ConfigurableApplicationContext, though..)

Reply:
I'll revert the logic error, but I think the instanceof part is suspicious. the pull request @bsideup did changed things besides instanceof and focusing on indexof check where there are many places in this code that call getBean still.. seems to draw attention to something that suggests everything has similar attention. anyway will revert this and hope we get to a point where instanceof is limiting us